job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -1364,6 +1364,8 @@ class PublisherContext extends AbstractExtensibleContext {
         publisherNodes << new NodeBuilder().'org.jenkinsci.plugins.postbuildscript.PostBuildScript' {
             buildSteps(context.stepContext.stepNodes)
             scriptOnlyIfSuccess(context.onlyIfBuildSucceeds)
+            scriptOnlyIfFailure(context.onlyIfBuildFails)
+            markBuildUnstable(context.markBuildUnstable)
```

Comment:
This breaks compatibility for everyone using an older version of the postbuildscript plugin. Before increasing the minimum version, there has to be a deprecation phase so that everyone has some time to migrate. See [this](https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.41/job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy#L1443-L1460) as an example for deprecating support for older plugin version. Also add corresponding entries to the [Migration](https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.41/docs/Migration.md) page and the [release notes](https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.41/docs/Home.md).